### PR TITLE
Use StorageNMap for Approvals in assets pallet

### LINKED ISSUE #8291: Implement StorageTripleMap
The assets pallet could use a `StorageTripleMap` to be event slightly more ergonomic:

https://github.com/paritytech/substrate/pull/8220#discussion_r589179259

This should be implemented, and the assets pallet should be updated to use it.

cc @thiolliere

### PR DESCRIPTION
**Pull request by KiChjang**
Fixes #8291.

### CODE REVIEW

File: frame/assets/src/lib.rs [hunk 1]
```diff
@@ -237,12 +237,13 @@ pub mod pallet {
...
-	pub(super) type Approvals<T: Config<I>, I: 'static = ()> = StorageDoubleMap<
+	pub(super) type Approvals<T: Config<I>, I: 'static = ()> = StorageNMap<
 		_,
-		Blake2_128Concat,
-		T::AssetId,
-		Blake2_128Concat,
-		ApprovalKey<T::AccountId>,
+		(
+			NMapKey<Blake2_128Concat, T::AssetId>,
+			NMapKey<Blake2_128Concat, T::AccountId>,
```
Comment (shawntabrizi) [inline frame/assets/src/lib.rs:244]: maybe nice to comment which account key is which both above and inline

**Review by shawntabrizi — APPROVED** [review]

**Review by shawntabrizi — APPROVED** [review]
looks right to me :)

need to make sure @jacogr signs off on this before we actually merge

**Comment by jacogr** [comment]
I have holes isn the JS API atm - so stuff like the `.keys` and `.entries` iterator have not been done for NMap yet. (Generally prefer to have something in-storage first to be able to generate proper test-cases for e2e and actual offline/static, this provides that. Additionally, also thinking, as you are, of just ripping out stuff and making Map/DoubleMap/NMap use all the same stuff, e.g. combining underneath the hood)

In this case it is purely the approvals for which I don't yet have something UI-wise. Additionally know of nothing out there tool, UI-wise or otherwise that uses this functionality atm. So it is technically non-breaking for existing tools. Even explorers need work anyway before they are able to index chains using this (e.g. the Python libs and Ruby libs)

Need to get the first items sorted anyway API-wise before being able to do anything. (It is "just work", nicely marked, so no crisis - just need a gap)

TL:DR Get it in.

**Review by jacogr — APPROVED** [review]

**Review by gavofyork — APPROVED** [review]

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**